crypto/src/main/scala/coop/rchain/crypto/signatures/Secp256k1.scala

```diff
@@ -1,9 +1,45 @@
 package coop.rchain.crypto.signatures
 
+import coop.rchain.crypto.codec.Base16
+
+import java.security.{KeyPairGenerator, SecureRandom}
+import java.security.interfaces.ECPrivateKey
+import java.security.spec.ECGenParameterSpec
+
 import org.bitcoin._
+import org.bouncycastle.jce.provider.BouncyCastleProvider
 
 object Secp256k1 {
 
+  private val provider  = new BouncyCastleProvider()
+  private val curveName = "secp256k1"
+
+  /**
+    * Verifies the given secp256k1 signature in native code.
+    *
+    * {{{
+    * >>> import coop.rchain.crypto.hash.Sha256
+    * >>> val (sec, pub) = Secp256k1.newKeyPair
+    * >>> val data = Sha256.hash("testing".getBytes)
+    * >>> val sig = Secp256k1.sign(data, sec)
+    * >>> Secp256k1.verify(data, sig, pub)
+    * true
+    * }}}
+    *
+    * @return (private key, public key) pair
+    *
+    */
+  def newKeyPair: (Array[Byte], Array[Byte]) = {
```

Comment:
This signature is prone to errors. I would like to have a wrapper types to make sure the caller doesn't mix them.

I'm thinking something like this:

```
case class PublicKey(bytes:Array[Byte])
case class SecretKey(bytes:Array[Byte])

def newKeyPair : (PublicKey, SecretKey) = {...}
```

I can see that `Array[Byte]`S are used through the whole file but this is a small change with big long term benefits.

Reply:
This is consistent with the approach in all the crypto libraries we use. If we decide to refactor here we should do it in all other cryptos as well.

Reply:
Consistently unsafe is not good :)

The crypto libraries provide the implementation of the algorithms and, probably, choosed to return `Array[Byte]` for easier integration. I don't question that decision. 

In our case we need type safety more than we need easy integration. For this reason, on top of the actual bytes that represent the keys, we should add strong type information in order to eliminate a certain class of bugs.

Reply:
I agree with @odeac . Returning a pair of byte arrays is horrible, especially for a public method.